lib/SparqlGenerator.js

```diff
@@ -255,30 +256,34 @@ Generator.prototype.toExpression = function (expr) {
 
 // Converts the parsed entity (or property path) into a SPARQL entity
 Generator.prototype.toEntity = function (value) {
-  // regular entity
-  if (isString(value)) {
-    switch (value[0]) {
+  if (isTerm(value)) {
+    // Remove the prefix but with taking a copy to avoid altering the input
+    if (value.termType === "BlankNode") value = {termType: value.termType, value: value.value.replace(/^e_|^g_/, '')};
```

Comment:
Actually this was not necessary anymore for the tests. It was something I added to prevent e_g_e_g_... in a query that was repeatedly parsed and then generated again. But even that won't happen with the current implementation of the parser:
`if (name.startsWith("e_")) return Parser.factory.blankNode(name);`
Blanknodes after an arbitrary amount of parsing and generation will simply be something like _:e_g_7

Are you happy with this behaviour? If so I will remove this line :)

Reply:
Happy with this behavior 🙂